src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1667,7 +1667,8 @@ RouterInsertTaskList(Query *query, bool parametersInQueryResolved,
 		relationShard->relationId = distributedTableId;
 
 		modifyTask->relationShardList = list_make1(relationShard);
-		modifyTask->taskPlacementList = ShardPlacementList(modifyRoute->shardId);
+		modifyTask->taskPlacementList = ActiveShardPlacementList(
```

Comment:
This one already worked fine without this change, because later non active placements are filtered out.